sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/statsbeat/_statsbeat.py

```diff
@@ -53,19 +84,33 @@ def collect_statsbeat_metrics(exporter) -> None:
                 long_interval_threshold,
                 exporter._credential is not None,
                 exporter._distro_version,
-            )
-        # Export some initial stats on program start
+            )        # Export some initial stats on program start
         mp.force_flush()
         # initialize non-initial stats
         _STATSBEAT_METRICS.init_non_initial_metrics()
 
+        # Schedule a second export after the initial delay to send feature, instrumentation,
+        # and attach statsbeat metrics (which have a 15-second delay)
+        timer = Timer(_INITIAL_DELAY_SECONDS, _delayed_export_statsbeat)
+        timer.daemon = True  # Set as daemon so it doesn't block program exit
+        timer.start()
 
 def shutdown_statsbeat_metrics() -> None:
+    """
+    Shut down the statsbeat metrics collection system.
+    
+    This ensures proper cleanup of resources and marks the system as shut down
+    to prevent further metric collection.
+    """
     global _STATSBEAT_METRICS
     shutdown_success = False
     if _STATSBEAT_METRICS is not None:
         with _STATSBEAT_LOCK:
             try:
+                # Cancel any pending timers by marking for shutdown
```

Comment:
You don't need to mark for shutdown pre-emptively if you are using STATSBEAT_LOCK at the top already for delayed shutdown.

Reply:
Makes sense, removed the unneeded shutdown logic.